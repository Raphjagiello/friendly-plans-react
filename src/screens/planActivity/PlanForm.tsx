import { Formik, FormikProps } from 'formik';
import React from 'react';
import { NavigationInjectedProps, withNavigation } from 'react-navigation';

import { ModelSubscriber, Plan } from 'models';
import { PlanFormControls } from './PlanFormControls';

export interface PlanFormData {
  planInput: string;
}

interface State {
  plan: Plan;
}

const FORM_INITIAL_VALUES: PlanFormData = {
  planInput: '',
};

export class PlanForm extends React.PureComponent<NavigationInjectedProps, State> {
  plansSubscriber: ModelSubscriber<Plan> = new ModelSubscriber();

  state: State = {
    plan: new Plan(),
  };

  async setPlanState(planInput: string) {
    const { id } = this.props.navigation.getParam('student');

    const plan = await Plan.create(id, planInput);

    this.setState({ plan });
  }

  onSubmit = ({ planInput }: PlanFormData) => this.setPlanState(planInput);

  renderFormControls = (props: FormikProps<PlanFormData>) => <PlanFormControls {...props} plan={this.state.plan} />;

  render() {
    return <Formik initialValues={FORM_INITIAL_VALUES} onSubmit={this.onSubmit} render={this.renderFormControls} />;
  }
}

export default withNavigation(PlanForm);
