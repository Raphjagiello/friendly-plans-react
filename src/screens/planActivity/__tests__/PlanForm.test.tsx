import { shallow } from 'enzyme';
import React from 'react';

import { navigationMock } from 'helpers';
import { PlanForm } from '../PlanForm';

describe('PlanFormControls', () => {
  const props = {
    navigation: navigationMock,
  };

  it('renders correctly', () => {
    const planInput = shallow(<PlanForm {...props} />);

    expect(planInput).toMatchSnapshot();
  });
});
