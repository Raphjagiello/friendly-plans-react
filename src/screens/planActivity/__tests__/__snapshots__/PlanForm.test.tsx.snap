// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlanFormControls renders correctly 1`] = `
<Formik
  enableReinitialize={false}
  initialValues={
    Object {
      "planInput": "",
    }
  }
  isInitialValid={false}
  onSubmit={[Function]}
  render={[Function]}
  validateOnBlur={true}
  validateOnChange={true}
/>
`;
